feat: add point shapes

markov00 opened the pull request:
## Summary

This PR adds the ability to choose the shape of the rendered point as per request for Timelion replacement.
<img width="928" alt="Screenshot 2021-01-20 at 18 28 37" src="https://user-images.githubusercontent.com/1421091/105212155-5f098180-5b4d-11eb-9d75-d7b88d00ae62.png">
 

fix #987 

### Checklist

Delete any items that are not applicable to this PR.

- [x] Any consumer-facing exports were added to `src/index.ts` (and stories only import from `../src` except for test data & storybook)
- [x] Proper documentation or storybook story was added for features that require explanation or tutorials
- [x] Unit tests were updated or added to match the most common scenarios

rshen91 reviewed (approved):
LGTM - tested locally on my mac. I like how this is set up to easily add other point shapes in the future in the theme. Defaulting to circle makes a lot of sense. 👍

nickofthyme reviewed (approved):
LGTM. I wonder how we could change the highlighter in the future based on the point shape type so we don't get this...

![Screen Recording 2021-01-21 at 10 22 AM](https://user-images.githubusercontent.com/19007109/105379607-9ec67c80-5bd2-11eb-98dc-230eca652728.gif)

monfera reviewed (approved):
LGTM 🎉

markov00 commented:
> LGTM. I wonder how we could change the highlighter in the future based on the point shape type so we don't get this...
> 
> ![Screen Recording 2021-01-21 at 10 22 AM](https://user-images.githubusercontent.com/19007109/105379607-9ec67c80-5bd2-11eb-98dc-230eca652728.gif)

@nickofthyme @monfera @rshen91  I've updated the PR after Nick's comment here.
I've matched the highlighter shape with the point shape.
I've also cleaned up the canvas rendering functions and created a set of reusable SVG paths.
We have now the ability to display also bubble charts with different shape sizes like:
<img width="660" alt="Screenshot 2021-01-27 at 10 25 35" src="https://user-images.githubusercontent.com/1421091/105971325-ada5a700-608a-11eb-8559-bb8f547e08e2.png">
 or having a single series with customized shapes per point:
<img width="526" alt="Screenshot 2021-01-27 at 10 31 10" src="https://user-images.githubusercontent.com/1421091/105971426-cb730c00-608a-11eb-999a-d0b2f74fb409.png">

I've also introduced a change on the current PointGeometry, computing the style override at the same time when computing the point position and scales, instead of computing it in the renderer.

monfera commented:
With the broad generalization introduced with the last commits, we, at some point, might want to move away from singleton legends, because each aesthetic channel may encode different categorical or quantitative data. Quick example:
![image](https://user-images.githubusercontent.com/1548516/106002036-f1aba280-60b0-11eb-8430-81a15576833e.png)

(the size legend is rendered somewhat useless though, due to lack of decent comparability of sizes)

While the marker shape may be redundant encoding for line series, eg. for accessibility and discernibility in print (at the cost of making the lines noisier), scatterplots and connected scatterplots would be more inviting for independent channel assignment. 

So with that, the question is, what are the tradeoffs of introducing broad configurability of markers as an independent categorical-only channel with or without independent legends. Without this, the ability to vary markers may not be super useful, though users can just avoid configuring it.

monfera commented on src/utils/themes/theme.ts:
```diff
@@ -306,6 +308,18 @@ export type DisplayValueStyle = Omit<TextStyle, 'fill' | 'fontSize'> & {
+  Circle: 'circle' as const,
+  Square: 'square' as const,
+  Diamond: 'diamond' as const,
+  Cross: 'cross' as const,
```
An oft used name is, eg. listed in [ggplot2](http://www.sthda.com/english/wiki/r-plot-pch-symbols-the-different-point-shapes-available-in-r), for `+`: plus; for `x`: cross

monfera reviewed (approved):
Good functional addition; asking for renaming "angled cross" to something standard, eg. cross, while renaming current cross to plus. Also, a couple of follow-up issues are warranted: 
- linking back to the independent legend point with examples
- using, or making it configurable, the markers in the legend, if redundant encoding is chosen

nickofthyme reviewed (approved):
LGTM. Nice job with the highlighter shapes, that looks much better!

I wonder if in the future we allow the user to pass a path, if we could scale that up and take the outline silhouette of the shape as the highlight geometry rather than just hardcoding each outlined shape. Do you think that would be possible for most shapes?

markov00 commented:
:tada: This PR is included in version 24.5.0 :tada:

The release is available on:
- [GitHub release](https://github.com/elastic/elastic-charts/releases/tag/v24.5.0)
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@elastic/charts/v/24.5.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
